plugins/hls-notes-plugin/README.md

```diff
@@ -0,0 +1,30 @@
+# Note plugin
+
+The [Note convention](https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/coding-style#2-using-notes) is a nice way to hoist and share big chunks of documentation out of the body of functions. This is done by referencing a long form note from within the function. This plugin extends goto-definition to jump from the reference to the note.
+
+# Example
+
+Main.hs
+```haskell
+module Main where
+
+main :: IO
+main = do
+    doSomething -- We need this here, see Note [Do Something] in Foo
+```
```

Comment:
Seems, both `@Note [Do Something]@` and `Note [Do Something]` are supported, right?

The documentation likely should reflect this!